vpr/src/route/router_lookahead_extended_map.cpp

```diff
@@ -606,10 +606,10 @@ float ExtendedMapLookahead::get_expected_cost(
 
 #ifndef VTR_ENABLE_CAPNPROTO
 
-void ExtendedMapLookahead::read(const std::string& file) {
+void ExtendedMapLookahead::read(const std::string& /*file*/) {
     VPR_THROW(VPR_ERROR_ROUTE, "MapLookahead::read not implemented");
 }
-void ExtendedMapLookahead::write(const std::string& file) const {
+void ExtendedMapLookahead::write(const std::string& /*file*/) const {
```

Comment:
Instead of having two method definitions and disabling one of them, we can write a single definition and use `#ifdef` to disable specific parts of the code within the function body. We can suppress unused argument warnings by employing the `[[maybe_unused]]` attribute.

Reply:
That seems like a good idea to me. @AlexandreSinger  what do you think?

Reply:
I agree completely with putting the IFDEFs within the function declarations, as this would help prevent regressions. I can transition the code in this file to do that.

I do not think we should use the [[maybe_unused]] attribute. We still want to know if an argument goes unused when we do not intend it to. We can simply just void the arguments within the function like this:
```
void foo(int arg1, int arg2) {
#IFNDEF
  // Original implementation
#ELSE
  (void)arg1;
  (void)arg2;
  VPR_THROW(VPR_ERROR_ROUTE, "...");
#ENDIF
}
```

This way the arguments are explicitly thrown away and the warning will still be generated in the original implementation (signifying there is likely an error).

Reply:
Updated. @soheilshahrouz Let me know if this is what you had in mind!

Reply:
I think void casting is a better practice because we can catch cases where the parameter is actually ununsed.